src/plugins/profiles/VirialProfile.cpp

```diff
@@ -26,7 +26,15 @@ void VirialProfile::output (string prefix, long unsigned accumulatedDatasets) {
 	// y value, PD, PX, PY, PZ
 	outfile << "# y\tvn-vt\tpx\tpy\tpz\n# \n";
 
-	double layerVolume = _samplInfo.globalLength[0] * _samplInfo.globalLength[1];
+	unsigned long unID;
+	double layerVolume;
+    double layerHeight = _samplInfo.globalLength[1]/_samplInfo.universalProfileUnit[1];
+	if(_samplInfo.cylinder){
+	    layerVolume = layerHeight * M_PI * _samplInfo.globalLength[0]*_samplInfo.globalLength[0];
```

Comment:
should this be globalLength[0] * globalLength[0] or [0] and [2] or is it irrelevant? 
add a comment to explain

Reply:
Its height * pi * r^2 for the layerVolume in cylinder sampling. Will add comment.